### DISCUSSION TITLE
Table 组件的 ShowSearchButton 设置为 false 仍然显示搜索按钮

### DISCUSSION THREAD

**Discussion by tony26050369** [dotnetcore/BootstrapBlazor#3489] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

使用Table组件时，如果设置ShowSearchButton = false，仍然会显示出“搜索”按钮。

### Expected Behavior

不应该显示出搜索按钮

### Steps To Reproduce

用于重现的项目：https://github.com/tony26050369/BBIssues/
重现步骤：进入“/TableIssues" 页面，会看到仍然显示着搜索按钮，而代码中已经设置了ShowSearchButton = false。

我想简单解释一下为什么我们需要隐藏这个搜索按钮。因为我们的搜索条件比较麻烦，所以把搜索条件都放了SearchTemplate中，用户需要点击“高级搜索”进行查询，同时设置了ShowSearchText="false"，所以单独的这个搜索按钮是没有任何作用的。

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/23045461/015ea81d-c57a-42a0-a4c5-a4137cdf4b79)

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/23045461/1eb4be70-6184-4f20-a836-52b6a3e8abca)

### .NET Version

8

**Comment by bb-auto** [comment]
@tony26050369 Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by tony26050369** [comment]
> @tony26050369 Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

[https://github.com/tony26050369/BBIssues/](url)

**Reply by ArgoZhang** [reply]
`ShowSearchButton` 控制的是搜索弹窗的搜索按钮
`ShowSearch` 控制的是工具栏中的搜索组件

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/22001478/154e1d97-2d2e-454f-bfb1-8bbce9eb379d)

**Reply by tony26050369** [reply]
@ArgoZhang  Hi, Argo,

在8.2.X版本中，按照文档所说，设置ShowSearch=true, ShowSearchText=false，会仅显示“高级搜索”，这是正常的。

但是，升级到8.3.X后，按照以上设置，不仅显示了“高级搜索”，同时搜索按钮也会被显示出来的，仅是搜索的文本框被隐藏了。

**Reply by ArgoZhang** [reply]
我更新了文档，功能我看没问题，你创建 issue 你要写清楚你的目的啊。如果是文档不对，就指出是文档不对，如果是功能不对，就要描述清楚，通过怎样的设置，期望得到什么结果。实际是什么结果，你这描述了半天，根本不知道你要表达什么

**Comment by tony26050369** [comment]
@ArgoZhang  Hi, Argo,

抱歉，我没有把我的问题讲清楚。

我想问一下，在搜索栏中，如何只保留“高级搜索”按钮，将其他的按钮隐藏掉？如下图。以前在8.2.版本中，只需要设置ShowSearchButton="false"，就可以把这三个按钮隐藏掉。从8.3版本之后，就不行了。谢谢！

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/23045461/c456795b-e17c-4ff8-aa12-1df9d81ee641)

**Reply by ArgoZhang** [reply]
@tony26050369 目前设计上无法只保留高级搜索

![6656b8553192918338e840c8aacabbc1](https://github.com/dotnetcore/BootstrapBlazor/assets/22001478/3e8a80d6-509b-4520-b3b4-f329d893a22c)

**Reply by tony26050369** [reply]
@ArgoZhang Hi, 这个需求对我们多个项目挺重要的，而且造成我们所有的项目停留在8.2版无法升级，能不能考虑加上这个设置？

谢谢

**Reply by ArgoZhang** [reply]
我看看能否再加一个参数控制这个按钮。实在是没有直观的名字可以用了

**Reply by ArgoZhang** [reply, marked as answer]
临时发了一个包 `8.5.8-beta01` `ShowSearchText="false"` 时那个搜索按钮也一起不显示，跟 `8.2.0` 行为一致，但是你这样用功能是少了一个啊

举例：用户设置了高级搜索条件，点击高级搜索弹窗中的搜索按钮，返回表格，表格显示数据。其他同事有数据操作，此时需要重新搜索，用户只能重新点开高级搜索弹窗，再次点击弹窗中的搜索按钮，才可以获得新数据，非常繁琐

**Reply by tony26050369** [reply]
@ArgoZhang  Hi, Argo,

谢谢帮助，我按照你的提示升级到了8.5.8-beta03，设置了ShowSearchText="false" ，但是搜索按钮仍然是显示出来的，如下图。重现的项目已经上传到：https://github.com/tony26050369/BBIssues

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/23045461/8d524298-fd50-447b-9e89-0fdf6f5f6c9f)

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/23045461/7980f11d-e08d-402a-9a5e-2f7855bd51a8)

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/23045461/7d8b80f4-6985-4273-987d-5236a0dbc8af)


从操作上讲，确实存在你说的问题，用户每次搜索都需要点击“高级搜索”打开对话框，然后再进行搜索，确实比较麻烦。但是，我确实想不出其他更好的办法了，因为我们项目的搜索条件基本上都是结构化的数据，不能进行模糊查询，如下图：

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/23045461/df895469-0bb0-4f9c-a840-ebc698158d78)

**Reply by ArgoZhang** [reply]
我明明写的是 `8.5.8-beta01` 还是临时给你发的。你不要更新到 `beta03` 啊，你的需求是你的特殊需求，并不代表用组件的人都你这样用啊。至少我是一定要那个搜索按钮的，不然太麻烦了啊。

我考虑了一下。你自己的项目里写个样式，把那个搜索按钮隐藏掉。就仅剩下高级搜索按钮了
